### ISSUE TITLE
Joint analysis: Cannot convert data from type LIST_ARRAY (<class 'list'>) to type VECTOR_VECTOR_REAL

### ISSUE THREAD

**Issue by miguelgcasado** [sertansenturk/tomato#87]
Hello,

while running the notebook of joint_analysis_demo and doing the predominant melody extraction I am having an error coming from essentia library.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
~/anaconda3/lib/python3.6/site-packages/essentia/standard.py in compute(self, *args)
     91                 try:
---> 92                     convertedData = _c.convertData(arg, goalType)
     93                 except TypeError:

~/anaconda3/lib/python3.6/site-packages/essentia/common.py in convertData(data, goalType)
    330     raise TypeError('Cannot convert data from type %s (%s) to type %s' %
--> 331                     (str(origType), str(type(data)), str(goalType)))
    332 

**TypeError: Cannot convert data from type LIST_ARRAY (<class 'list'>) to type VECTOR_VECTOR_REAL**

During handling of the above exception, another exception occurred:

TypeError                                 Traceback (most recent call last)
<ipython-input-5-63782434cc02> in <module>
      4 
      5 # predominant melody extraction
----> 6 audio_pitch = audioAnalyzer.extract_pitch(audio_filename)
      7 # NOTE: do not call pitch filter later as aligned_pitch_filter will be more effective

~/anaconda3/lib/python3.6/site-packages/tomato/audio/audioanalyzer.py in extract_pitch(self, filename)
    212                     format(filename))
    213 
--> 214         results = self._pitch_extractor.run(filename)
    215         pitch = results['settings']  # collapse the keys in settings
    216         pitch['pitch'] = results['pitch']

~/anaconda3/lib/python3.6/site-packages/tomato/audio/predominantmelody.py in run(self, fname)
     88 
     89         contours_bins, contours_start_times, contour_saliences, duration = \
---> 90             self._extract_pitch_contours(audio)
     91 
     92         # run the simplified contour selection

~/anaconda3/lib/python3.6/site-packages/tomato/audio/predominantmelody.py in _extract_pitch_contours(self, audio)
    172             run_pitch_contours(
    173                 pool['allframes_salience_peaks_bins'],
--> 174                 pool['allframes_salience_peaks_contourSaliences'])
    175         return contours_bins, contours_start_times, contour_saliences, duration
    176 

~/anaconda3/lib/python3.6/site-packages/essentia/standard.py in __call__(self, *args)
    116 
    117         def __call__(self, *args):
--> 118             return self.compute(*args)
    119 
    120         def __str__(self):

~/anaconda3/lib/python3.6/site-packages/essentia/standard.py in compute(self, *args)
     93                 except TypeError:
     94                     raise TypeError('Error cannot convert argument %s to %s' \
---> 95                           %(str(_c.determineEdt(arg)), str(goalType)))
     96 
     97                 convertedArgs.append(convertedData)

**TypeError: Error cannot convert argument LIST_ARRAY to VECTOR_VECTOR_REAL**

Probably it's an trivial error related with the loading of the file, or the installation of essentia, but i can not solve it.
Thanks for your help!

**Comment by sertansenturk** [comment]
Thank you Miguel for reporting the issue. 

I think this is probably because of the Essentia version, considering that a lot of time has passed since I have tested tomato an up-to-date Essentia version.

Can you tell me?
- Your OS? (Ubuntu 16.04, Mac OSX Sierra etc.)
- Essentia version? You can check it by:
```python
from essentia import __version__ as es_version
print(es_version)
```

I addition can you also compile [2.1 beta4](https://essentia.upf.edu/documentation/installing.html#compiling-essentia) and report whether the problem persists?

I am typically busy on the weekdays due to my day job, so I will try to get back to this issue next Sunday. I hope it is OK.

**Comment by miguelgcasado** [comment]
My OS is Ubuntu 18.04 and the essentia version is 2.1-beta5. The error is clearly getting the pitch in this section: 
```
~/anaconda3/lib/python3.6/site-packages/tomato/plotter.py in _parse_pitch(pitch_in)
    110         print(pitch)
    111         pitch = np.array(pitch)  # convert to numpy array
--> 112         pitch[pitch[:, 1] < 20.0, 1] = np.nan  # remove inaudible for plots
    113 
    114         return pitch

IndexError: too many indices for array
```

Because then, when I print the file information the output looks like that: 
```
{'makam': 'ussak',
 'melodic_progression': None,
 'metadata': {'mbid': 'f970f1e0-0be9-4914-8302-709a0eac088e',
  'path': 'ussak--sazsemaisi--aksaksemai----neyzen_aziz_dede/f970f1e0-0be9-4914-8302-709a0eac088e/f970f1e0-0be9-4914-8302-709a0eac088e.mp3',
  'duration': 223.17939479239976,
  'sampling_frequency': 44100,
  'bit_rate': 160,
  'url': 'http://musicbrainz.org/recording/f970f1e0-0be9-4914-8302-709a0eac088e',
  'title': 'Uşşak Sazsemaisi',
  'releases': [{'title': 'Tanburi Cemil Bey - Volumes II & III',
    'mbid': '805a3604-92e6-482f-a0e3-6620c4523d7a'}],
  'artist_credits': [{'name': 'Tanburi Cemil Bey',
    'mbid': '4cc5200e-a3f7-4d5d-8b15-9b10a8f9a9a2'}],
  'artists': [{'name': '[unknown]',
    'mbid': '125ec42a-7229-4250-afc5-e057484327fe',
    'type': 'instrument',
    'attribute-list': ['tanbur']},
   {'name': '[unknown]',
    'mbid': '125ec42a-7229-4250-afc5-e057484327fe',
    'type': 'instrument',
    'attribute-list': ['cello']},
   {'name': 'Tanburi Cemil Bey',
    'mbid': '4cc5200e-a3f7-4d5d-8b15-9b10a8f9a9a2',
    'type': 'instrument'}],
  'works': [{'title': 'Uşşak Saz Semaisi',
    'mbid': 'e7924b0d-c8a0-4b4a-b253-8eec898eac1e'}],
  'makam': [{'mb_attribute': 'Uşşak',
    'attribute_key': 'ussak',
    'source': 'http://musicbrainz.org/work/e7924b0d-c8a0-4b4a-b253-8eec898eac1e'},
   {'mb_tag': 'uşşak',
    'attribute_key': 'ussak',
    'source': 'http://musicbrainz.org/recording/f970f1e0-0be9-4914-8302-709a0eac088e'}],
  'form': [{'mb_attribute': 'Sazsemaisi',
    'attribute_key': 'sazsemaisi',
    'source': 'http://musicbrainz.org/work/e7924b0d-c8a0-4b4a-b253-8eec898eac1e'},
   {'mb_tag': 'sazsemaisi',
    'attribute_key': 'sazsemaisi',
    'source': 'http://musicbrainz.org/recording/f970f1e0-0be9-4914-8302-709a0eac088e'}],
  'usul': [{'mb_attribute': 'Aksaksemai',
    'attribute_key': 'aksaksemai',
    'source': 'http://musicbrainz.org/work/e7924b0d-c8a0-4b4a-b253-8eec898eac1e'},
   {'mb_tag': 'aksaksemai',
    'attribute_key': 'aksaksemai',
    'source': 'http://musicbrainz.org/recording/f970f1e0-0be9-4914-8302-709a0eac088e'}],
  'instrumentation_voicing': 'trio instrumental'},
 'note_models': None,
 'pitch': None,
 'pitch_class_distribution': None,
 'pitch_distribution': None,
 'pitch_filtered': None,
 'tempo': None,
 'tonic': None,
 'transposition': None}
```
This is not an urgent issue, so whenever you find the time to help me will be good!

**Comment by sertansenturk** [comment]
Closed by #88
